transformer_engine/pytorch/attention.py

```diff
@@ -795,10 +797,20 @@ def __init__(
         norm_factor = math.sqrt(self.hidden_size_per_attention_head)
 
         self.device_compute_capability = get_device_compute_capability()
+        self.deterministic = not bool(int(os.getenv("NVTE_ALLOW_NONDETERMINISTIC_ALGO", "1")))
+
         self.use_flash_attention = (
             int(os.getenv("NVTE_FLASH_ATTN", "1"))
             and self.device_compute_capability >= 8.0
         )
+        if _flash_attn_2_available and self.deterministic:
+            self.use_flash_attention = False
+            warnings.warn(
+                "Disabling usage of FlashAttention since version 2 does not support deterministic"
+                "exection. In order to use FA with deterministic behavior, install FlashAttention"
+                "version 1."
+            )
```

Comment:
Maybe add after L967 in your attention.py:
`if fused_attention_backend == FusedAttnBackend["F16_arbitrary_seqlen"] and self.deterministic:
    use_fused_attention = False`